src/Propel/Generator/Model/Database.php

```diff
@@ -608,36 +627,44 @@ public function getSchemaDelimiter()
      */
     public function setSchema($schema)
     {
-        $oldSchema = $this->schema;
-        if ($this->schema !== $schema && $this->getPlatform()) {
-            $schemaDelimiter = $this->getPlatform()->getSchemaDelimiter();
-            $fixHash = function (&$array) use ($schema, $oldSchema, $schemaDelimiter) {
-                foreach ($array as $k => $v) {
-                    if ($schema && $this->getPlatform()->supportsSchemas()) {
-                        if (false === strpos($k, $schemaDelimiter)) {
-                            $array[$schema . $schemaDelimiter . $k] = $v;
-                            unset($array[$k]);
-                        }
-                    } elseif ($oldSchema) {
-                        if (false !== strpos($k, $schemaDelimiter)) {
-                            $array[explode($schemaDelimiter, $k)[1]] = $v;
-                            unset($array[$k]);
-                        }
-                    }
-                }
-            };
+        if (!($this->schema !== $schema && $this->getPlatform())) {
```

Comment:
It became unnecessarily complicated for reading. 
Please open brackets or leave a comment.

Reply:
I don't understand what you mean by opening brackets. Do you want me to just simplify the boolean statement with demorgan's law like this?

```suggestion
        if ($this->schema === $schema && !$this->getPlatform()) {
```